docs/adr/001_error_handling.md

```diff
@@ -0,0 +1,107 @@
+1. Devolves into error type per method of a particular trait type (e.g., `SdkShutdownResult`, `SdkExportResult`) _if the error types need to diverge_
+1. May alternatively devolve into error type per signal (e.g., `SpanExporter`) if the _signals diverge_
+
+This approach generalises across both **signals** and **trait methods**. For example, returning to our exporter traits, we have a trait that looks the same for each signal, with the same three methods. Upon closer inspection ([#2600](https://github.com/open-telemetry/opentelemetry-rust/issues/2600)), the potential error set is the same both between the methods *and* between the signals; this means we can use a single shared error type across both axes:
+
+```rust
+
+#[derive(Error, Debug)]
+
+// Errors that can occur during SDK operations export(), force_flush() and shutdown().
+pub enum OTelSdkError {
+
+    /// All error types in here may be returned by any of the operations
+    /// of the exporters, on any of their APIs.
+    /// If this were not the case, we would not be able to use a shared error.
+    #[error("Shutdown already invoked")]
+    AlreadyShutdown,
+
+    // ... Other errors ...
+
+    /// The error message is intended for logging purposes only and should not
+    /// be used to make programmatic decisions. It is implementation-specific
+    /// and subject to change without notice. Consumers of this error should not
+    /// rely on its content beyond logging.
+    #[error("Operation failed: {0}")]
+    InternalFailure(String),
+}
+
+pub type OTelSdkResult = Result<(), OTelSdkError>;
+```
+
+... which the traits themselves use:
+
+```rust
+
+//
+// The actionable errors returned by the exporter traits are effectively
+// the same for all operations; we can use a single shared error.
+// 
+
+use opentelemetry_sdk::error::OTelSdkResult;
+
+pub trait LogExporter {
+	fn export(...) -> OtelSdkResult;
+	fn shutdown(...) -> OtelSdkResult; 
+  fn force_flush(...) -> OTelSdkResult;
+}
+
+// ...
+
+pub trait SpanExporter {
+	fn export(...) -> OtelSdkResult;
+	fn shutdown(...) -> OtelSdkResult;
+  fn force_flush(...) -> OTelSdkResult;
+}
+
+```
+
+### When to box custom errors
+
+Note above that we do not box anything into `InternalFailure`. Our rule here is that if the caller cannot reasonably be expected to handle a particular error variant, we will use a simplified interface that returns only a descriptive string. In the concrete example we are using with the exporters, we have a [strong signal in the opentelemetry-specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#export) that indicates concretely that the error types are not actionable by the caller.
+
+If the caller may potentially recover from an error, we will follow [canonical's rust best practices](https://canonical.github.io/rust-best-practices/error-and-panic-discipline.html) and instead preserve the nested error.
```

Comment:
I'd suggest to remove the link to canonical guides. It is not clear whether we'll always follow it or not.